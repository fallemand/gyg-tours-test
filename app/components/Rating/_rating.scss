@import '../../styles/variables';

.rating {
  padding: 5px 0;

  .rating__stars {
    display: inline-block;
    margin-right: 10px;
  }

  .rating__star {
    color: #e8e9ec;
    margin-right: 4px;

    &.rating__star--filled {
      color: #ffc600;
    }
  }

  .rating__info {
    font-size: 14px;
    color: $font-color-secondary;

    small {
      color: $font-color-secondary;
      font-size: $font-size-12;
    }
  }
}
