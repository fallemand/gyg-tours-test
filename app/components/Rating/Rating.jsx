import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import './_rating.scss';

const Rating = ({ className, value }) => (
  <div className={classnames(className, 'rating')}>
    <div className="rating__stars">
      {Array.from({ length: 5 }, (x, i) => i + 1).map(number => (
        <span
          key={number}
          className={classnames('rating__star', { 'rating__star--filled': (number <= value) })}
        >
            &#9733;
        </span>
      ))}
    </div>
    <span className="rating__info">
      {value}
      <small> /5</small>
    </span>
  </div>
);


Rating.propTypes = {
  className: PropTypes.string,
  value: PropTypes.number.isRequired,
};

Rating.defaultProps = {
  className: null,
};

export default Rating;
