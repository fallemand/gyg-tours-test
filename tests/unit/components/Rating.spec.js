import React from 'react';
import Rating from '../../../app/components/Rating';

describe('Rating', () => {
  const props = {
    value: 3.2,
    className: '__CLASSNAME__',
  };

  it('render component', () => {
    const component = shallow(<Rating {...props} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });

  it('should have 5 stars', () => {
    const component = shallow(<Rating {...props} />);
    const stars = component.find('.rating__star');
    expect(stars).toHaveLength(5);
  });

  it('count filled stars', () => {
    const component = shallow(<Rating {...props} />);
    let stars = component.find('.rating__star--filled');
    expect(stars).toHaveLength(3);
    component.setProps({
      value: 4.5,
    });
    stars = component.find('.rating__star--filled');
    expect(stars).toHaveLength(4);
  });
});
