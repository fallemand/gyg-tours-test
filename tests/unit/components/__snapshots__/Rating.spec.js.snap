// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating render component 1`] = `
"<div className='__CLASSNAME__ rating'>
  <div className='rating__stars'>
    <span className='rating__star rating__star--filled'>
      ★
    </span>
    <span className='rating__star rating__star--filled'>
      ★
    </span>
    <span className='rating__star rating__star--filled'>
      ★
    </span>
    <span className='rating__star'>
      ★
    </span>
    <span className='rating__star'>
      ★
    </span>
  </div>
  <span className='rating__info'>
    3.2
    <small>
       /5
    </small>
  </span>
</div>"
`;
